src/odemis/driver/orsay.py

```diff
@@ -0,0 +1,880 @@
+        else:  # if parameter.Actual is VALVE_TRANSIT, or undefined
+            logging.debug(log_msg % valve_state)
+
+    def _updatePosition(self, parameter=None, attributeName="Actual"):
+        """
+        parameter (Orsay Parameter): the parameter on the Orsay server to use to update the VA
+        attributeName (str): the name of the attribute of parameter which was changed
+
+        Reads the vacuum state from the Orsay server and saves it in the position VA
+        """
+        if parameter is None:
+            parameter = self._chamber.VacuumStatus
+        if parameter is not self._chamber.VacuumStatus:
+            raise ValueError("Incorrect parameter passed to _updatePosition. Parameter should be "
+                             "datamodel.HybridPlatform.AnalysisChamber.VacuumStatus. Parameter passed is %s"
+                             % parameter.Name)
+        if parameter.Actual == parameter.Target:
+            logging.debug("Target vacuum state reached.")
+            self._vacuumStatusReached.set()
+        else:
+            self._vacuumStatusReached.clear()
+        if attributeName != "Actual":
+            return
+        currentVacuum = int(parameter.Actual)
+        logging.debug("Vacuum status changed to %f." % currentVacuum)
+        self.position._set_value({"vacuum": currentVacuum}, force_write=True)
+
+    def _updatePressure(self, parameter=None, attributeName="Actual"):
+        """
+        parameter (Orsay Parameter): the parameter on the Orsay server to use to update the VA
+        attributeName (str): the name of the attribute of parameter which was changed
+
+        Reads the chamber pressure from the Orsay server and saves it in the pressure VA
+        """
+        if parameter is None:
+            parameter = self._chamber.Pressure
+        if parameter is not self._chamber.Pressure:
+            raise ValueError("Incorrect parameter passed to _updatePressure. Parameter should be "
+                             "datamodel.HybridPlatform.AnalysisChamber.Pressure. Parameter passed is %s"
+                             % parameter.Name)
+        if attributeName != "Actual":
+            return
+        self.pressure._set_value(float(parameter.Actual), force_write=True)
+
+    def _changeVacuum(self, goal, wait=True):
+        """
+        goal (int): goal state of the vacuum: (0: "vented", 1: "primary vacuum", 2: "high vacuum")
+        wait (bool): if True, the function will block until the goal vacuum state is reached
+        return (int): actual state of the vacuum at the end of this function: (0: "vented", 1: "primary vacuum",
+                      2: "high vacuum")
+
+        Sets the vacuum status on the Orsay server to argument goal and waits until it is reached.
+        Then returns the reached vacuum status.
+        """
+        logging.debug("Setting vacuum status to %s." % self.axes["vacuum"].choices[goal])
+        self._chamber.VacuumStatus.Target = goal
```

Comment:
To be extra careful, you should set `_vacuumStatusReached.clear()` before changing the target. Otherwise, if this code execute really fast, it might end up with the event still left in set() , and not wait.